### PR TITLE
Interactive functionality

### PR DESCRIPTION
**Pull request by axif0**
### Contributor checklist

- [X] This pull request is on a [separate branch](https://docs.github.com/en/get-started/quickstart/github-flow) and not the main branch

---

### Description
The [rich](https://github.com/Textualize/rich) library enhances CLI output with colorful tables and messages, while  [questionary](https://github.com/tmbo/questionary)  provides interactive prompts for user input.

attempt 2

### Related issue

- #203 
- #202
- #209

### CODE REVIEW

**Comment by axif0** [comment]
cc: @mhmohona, @andrewtavis if you don't like the interactive UI or if it is not user friendly for community, I can make another PR using  [asciimatics](https://github.com/peterbrittain/asciimatics) as @wkyoshida said.

File: src/scribe_data/cli/interactive.py [hunk 1]
```diff
@@ -23,168 +23,156 @@
...
+                    config.overwrite,
+                    config.output_type,
+                )
+                rprint(f"[green]✓[/green] Exported {language} {data_type} data.")
+
+    rprint(Panel.fit("[bold green]Data export completed successfully![/bold green]"))
+
+
+# MARK: Start interactive mode functionality
+def start_interactive_mode():
```
Comment (mhmohona) [inline src/scribe_data/cli/interactive.py:154]: Please add docstring here, as well as for other functions.

**Review by mhmohona — CHANGES_REQUESTED** [review]
Great work @axif0! ✨

**Comment by andrewtavis** [comment]
One question on this, @axif0: What's the support from the packages you're using to display options as a grid? I'm quite conscious right now that as soon as we start adding in the new languages that people are working on, that we won't be able to see all the options on the screen anymore without displaying a grid. Do you think this is an option for the current dependencies?

Also, don't we need dependencies added to the `requirements.txt` file?

**Comment by andrewtavis** [comment]
Sorry the dependencies are there... Lots of PRs over the last few days. I'm going to start testing this and comparing it against the work to re-ask users for options that was done in #216 and #239. I'm assuming that the dependencies will do this, but we'll see :)

**Review by andrewtavis — APPROVED** [review]
Ok lots of changes went through in my commit, but overall this is great, @axif0 😊

- Exit wasn't working
- We can now rerun data type only
- Minor color/style fixes
- Copy fixes (I changed export to request so it doesn't sound like the data is in Scribe-Data)
- Overall control fixes
- Asking the user if they want to continue when they didn't select something as the message was above the options for me, and is definitely going to be there for everyone as the options expand

This really is so cool :) :)

**Pull request merged**